workflows/denovotranscript.nf

```diff
@@ -28,13 +80,204 @@ workflow DENOVOTRANSCRIPT {
+    ch_filtered_reads = FASTQ_TRIM_FASTP_FASTQC.out.reads
+
+    if (params.remove_ribo_rna) {
+        ch_sortmerna_fastas = Channel.from(ch_ribo_db.readLines()).map { row -> file(row, checkIfExists: true) }.collect()
+        //
+        // MODULE: SORTMERNA
+        //
+        SORTMERNA (
+            ch_filtered_reads,
+            ch_sortmerna_fastas
+        )
+        ch_filtered_reads = SORTMERNA.out.reads
+        ch_multiqc_files = ch_multiqc_files.mix(SORTMERNA.out.log.collect{it[1]}.ifEmpty([]))
+        ch_versions = ch_versions.mix(SORTMERNA.out.versions)
+
+        //
+        // MODULE: FASTQC
+        //
+        FASTQC_FINAL (
+            SORTMERNA.out.reads
+        )
+        ch_multiqc_files = ch_multiqc_files.mix(FASTQC_FINAL.out.zip.collect{it[1]})
+        ch_versions = ch_versions.mix(FASTQC_FINAL.out.versions)
+    }
+
+    if (!params.QC_only) {
+
+        if (!params.skip_assembly) {
+
+            // All methods use pooled reads
+            ch_pool = ch_filtered_reads.collect { meta, fastq -> fastq }.map { [[id:'pooled_reads', single_end:false], it] }
+
+            //
+            // MODULE: CAT_FASTQ
+            //
+            CAT_FASTQ (
+                ch_pool
+            )
+            ch_versions = ch_versions.mix(CAT_FASTQ.out.versions)
+
+            ch_assemblies = Channel.empty()
+
+            if (params.trinity) {
+                //
+                // MODULE: TRINITY
+                //
+                TRINITY (
+                    CAT_FASTQ.out.reads
+                )
+                ch_versions = ch_versions.mix(TRINITY.out.versions)
+                ch_assemblies = ch_assemblies.mix(TRINITY.out.transcript_fasta)
+            }
+
+            if (params.trinity_no_norm) {
+                //
+                // MODULE: TRINITY_NO_NORM
+                //
+                TRINITY_NO_NORM (
+                    CAT_FASTQ.out.reads
+                )
+                ch_versions = ch_versions.mix(TRINITY_NO_NORM.out.versions)
+                ch_assemblies = ch_assemblies.mix(TRINITY_NO_NORM.out.transcript_fasta)
+            }
+
+            if (params.rnaspades) {
+                CAT_FASTQ.out.reads.map { meta, illumina ->
+                    [ meta, illumina, [], [] ] }.set { ch_spades }
+
+                //
+                // MODULE: SPADES
+                //
+                SPADES (
+                    ch_spades,
+                    [],
+                    []
+                )
+                ch_versions = ch_versions.mix(SPADES.out.versions)
+                ch_assemblies = ch_assemblies.mix(SPADES.out.transcripts)
+
+                if (params.soft_filtered_transcripts) {
+                    ch_assemblies = ch_assemblies.mix(SPADES.out.soft_filtered_transcripts)
+                }
+
+                if (params.hard_filtered_transcripts) {
+                    ch_assemblies = ch_assemblies.mix(SPADES.out.hard_filtered_transcripts)
+                }
+            }
```

Comment:
I think it would be more convenient to pass the different aligners here in a single comma separated list that you can then split into a list.

`--tools "rnaspades,trinity"`

```
def tools = params.tools.tokenize(',').

...
if (tools.contains("trinity")) {
    ....
}
```

This will scale better if you would later add more assemblers.

Reply:
thank you for the suggestion! I have added the `--assemblers` param to do this